python/triton/language/extern.py

```diff
@@ -58,9 +58,9 @@ def elementwise(lib_name: str, lib_path: str, args: list, arg_type_symbol_dict:
     dispatch_args = args.copy()
     all_scalar = True
     ret_shape = None
-    for i in range(len(dispatch_args)):
-        dispatch_args[i] = core._to_tensor(dispatch_args[i], _builder)
-        if dispatch_args[i].type.is_block():
+    for i, item in enumerate(dispatch_args):
+        item = core._to_tensor(item, _builder)
```

Comment:
This actually isn't right, we want to update the `dispatch_args` array. That's probably why I didn't use enumerate.

Reply:
I think it's probably fine to just use `range(len(...))` then...
We do have good reasons to use it at many places.

Reply:
yeah, I think the last commit isn't right. might some things , probably best to revert the changes

Reply:
@Jokeren can you run the testcases and review the PR now